### DISCUSSION TITLE
Change useRef value within a worklet

### DISCUSSION THREAD

**Discussion by rahulsyal2512** [software-mansion/react-native-reanimated#3198] [Q&A]
const x = useSharedValue(0);
  const offset = useRef(true);
  
  const gestureHandler = useAnimatedGestureHandler({
    onStart: (_, ctx) => {
      ctx.startX = x.value;
      offset.current = false     // <- how to use this here
    },
    onActive: (event, ctx) => {
      x.value = ctx.startX + event.translationX;
    },
    onEnd: _ => {
      x.value = withSpring(0);
      offset.current = true     // <-  how to use this here
    },
  });
 
  useRef value is not getting changed inside worklet.

  **Version**-
  react-native-reanimated - 2.4.0

**Comment by Abulie** [comment]
Hi, you can use another useSharedValue instead of useRef. There might be a better way to do this though...

Here's the part of the documentation:  

> "Use this hook to create a reference to a JavaScript value that can be shared with worklets"

> "...it can be accessed and modified by worklets. Shared Values can also be modified from the React Native thread directly, in which case the update is going to be asynchronous." 

https://docs.swmansion.com/react-native-reanimated/docs/api/hooks/useSharedValue/
